app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,31 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @user = User.includes(:posts).find(params[:user_id])
```

Comment:
Nice job so far👍🏼. As a requirement, you are meant to ensure the N+1 problem are solved when fetching all `posts` and their `comments` for a user😉. but what you're doing is fetching users with their associated posts😉. Please do well to see the suggestion below😉.

```suggestion
    @posts = @user.posts.includes(:comments)
```

I don't think you needed any other change in the show action😉.

Kindly note that you might have to adjust some of your tests when this correction is made, but ensure you stay consistent with the requirements and all tests pass 👍🏼.